subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -171,9 +171,9 @@ apply plugin: SomePlugin
                 public SubclassTask() {
                     // These methods changed in 3.2 in a backwards compatible way, but they
                     // leak internal types that cannot be moved without causing a breakage.
-                    getInputs().file("someFile");
-                    getInputs().files("anotherFile", "yetAnotherFile");
-                    getInputs().dir("someDir");
+                    getInputs().file("someFile").withPropertyName("input1");
+                    getInputs().files("anotherFile", "yetAnotherFile").withPropertyName("input2").withPathSensitivity(PathSensitivity.RELATIVE);
```

Comment:
I think we need to restrict this cross version test to something which supports `withPropertyName` and `withPathSensitivity` (3.1). The other parts of the test should work with earlier Gradle versions.
Or we only add calls to that if the previous Gradle version is at least 3.1.